### PR TITLE
[DOC] Add new section on Kafka Bridge

### PR DESCRIPTION
**Pull request by PaulRMellor**
The PR includes new content created for the download, deployment and configuration of the installation archive for the Kafka Bridge on RHEL.

This has been added along with the concepts Daniel has created for the Kafka Bridge on OCP.

Work in progress. Some of the content created for OCP may not be relevant here. As the OCP content changes, the changes should be reflected in the same sections here.

### CODE REVIEW

File: books/con-requests-kafka-bridge.adoc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+After creating a consumer, all subsequent requests must provide an `Accept` header in the following format:
+
+[source,http,subs=+quotes]
+----
+Accept: application/vnd.kafka._embedded-data-format_.v2+json
+----
+
+Where the `embedded-data-format` is the embedded data format of the consumer: either `json` or `binary`. The `+json` query parameter is not changed.
+
+For example, when committing a list of consumer offsets for a consumer that has an embedded data format of JSON, include this Accept header:
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply books/con-requests-kafka-bridge.adoc]: I have made the example more generic:

Here the embedded data format is `json`:

[source,http,subs=+quotes]
----
Accept: application/vnd.kafka.json.v2+json
----

File: books/con-overview-of-kafka-bridge.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-kafka-bridge.adoc
+
+[id='con-overview-of-kafka-bridge-{context}']
+= Overview of the {ProductName} Kafka Bridge
+
+The {ProductName} Kafka Bridge provides an API for integrating HTTP-based clients with a Kafka cluster running on {ProductPlatformName}. The API enables these clients to produce and consume messages without the requirement to use the native Kafka protocol.
+
+The API has two main resources -- `consumers` and `topics` -- that are exposed and made accessible through endpoints to interact with consumers and producers in your Kafka cluster. You can:
```
Comment (scholzj) [inline books/con-overview-of-kafka-bridge.adoc]: Do we need to clarify that these are related only to the HTTP Bridge it self and not to consumers connected directly to Kafka? I know that the Confluent HTTP Proxy has some admin features ... that makes we wonder if users might confuse this with them.

File: books/con-requests-kafka-bridge.adoc [hunk 3]
```diff
@@ -0,0 +1,100 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-kafka-bridge.adoc
+
+[id='con-requests-kafka-bridge-{context}']
+= Requests to the {ProductName} Kafka Bridge
+
+== Authentication and encryption
+
+Authentication and encryption between external clients and the Kafka Bridge is not yet supported. This means that requests sent from clients to the Kafka Bridge are:
```
Comment (scholzj) [inline books/con-requests-kafka-bridge.adoc]: I think this makes it more clear. `external` clients might not be completely easy to understand.

```suggestion
Authentication and encryption between HTTP clients and the Kafka Bridge is not yet supported. This means that requests sent from clients to the Kafka Bridge are:
```

File: books/con-requests-kafka-bridge.adoc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+* Not encrypted, and must use HTTP rather than HTTPS
+
+* Sent without authentication
+
+You can configure TLS or SASL-based authentication between the Kafka Bridge and your Kafka cluster.
+
+For more information, see xref:assembly-kafka-encryption-and-authentication-{context}[].
```
Comment (scholzj) [inline books/con-requests-kafka-bridge.adoc]: I do not think this is sufficient. That chapter only explains how to configure the authN/authZ in the broker. I think we need to clarify at least something like _You can configure the Bridge as any other Kafka client, just prefix the Kafka consumer/producer options with `kafka.`_

File: books/con-sending-requests-to-kafka-bridge.adoc [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-kafka-bridge.adoc
+
+[id='con-sending-requests-to-kafka-bridge-{context}']
+= Sending requests to the {ProductName} Kafka Bridge
+
+Content goes here.
+
+Include information on port-forwarding in this section because it's part of a Quickstart setup. Have a single procedure that includes setting up port-forwarding for local testing and development purposes only.
```
Comment (scholzj) [inline books/con-sending-requests-to-kafka-bridge.adoc]: Should this be updated for RHEL? Or just remove the whole concept?

File: books/proc-configuring-kafka-bridge.adoc [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+This procedure describes how to configure the properties used by the {ProductName} Kafka Bridge.
+
+.Prerequisites
+
+* xref:proc-installing-amq-streams-{context}[{ProductName} is installed on the host]
+* xref:proc-downloading-kafka-bridge-{context}[The Kafka Bridge installation archive is downloaded]
+
+.Procedure
+
+. Edit the the `configfile.properties` file provided with the {ProductName} Kafka Bridge installation archive.
```
Comment (scholzj) [inline books/proc-configuring-kafka-bridge.adoc]: I think the provided file is called `application.properties`? @ppatierno?

```suggestion
. Edit the the `application.properties` file provided with the {ProductName} Kafka Bridge installation archive.
```

File: books/proc-configuring-kafka-bridge.adoc [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
++
+Use the properties file to specify Kafka and HTTP-related properties.
+
+.. Configure standard Kafka-related properties, including properties specific to the Kafka consumers and producers.
```
Comment (scholzj) [inline books/proc-configuring-kafka-bridge.adoc]: I think we should be more clear about:
* Consumer options should be prefixed by `kafka.consumer.`
* Producer options should be prefixed by `kafka.producer.`
* General options prefixed with `kafka.` and will be overwritten if the options is specified with consumer or producer prefix.

File: books/proc-configuring-kafka-bridge.adoc [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
++
+---
+Use:
```
Comment (scholzj) [inline books/proc-configuring-kafka-bridge.adoc]: We should make it clear that this is just an example and not a list of supported options.

```suggestion
For example use:
```

File: books/proc-configuring-kafka-bridge.adoc [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+* `kafka.bootstrap.servers` to define the host/port connections to the Kafka cluster
+* `kafka.producer.acks` to provide acknowledgments to the HTTP client
+* `kafka.consumer.auto.offset.reset`’ to determine how to manage reset of the offset in Kafka
+---
++
+For more information on configuration of Kafka properties, see the http://kafka.apache.org[Apache Kafka website^]
```
Comment (scholzj) [inline books/proc-configuring-kafka-bridge.adoc]: We have in appendixes the list of all consumer and producer options. Should we link there? EIther instead of next to it?

File: books/proc-downloading-kafka-bridge.adoc [hunk 10]
```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// assembly-getting-started.adoc
+
+[id='proc-downloading-kafka-bridge-{context}']
+
+= Downloading an AMQ Streams Archive
```
Comment (scholzj) [inline books/proc-downloading-kafka-bridge.adoc]: This sounds a bit strange since using the _Archive_ alone gives me the sense that it's old. Should we say _ZIP archive_?

```suggestion
= Downloading an AMQ Streams ZIP Archive
```

File: books/proc-downloading-kafka-bridge.adoc [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+An archived distribution of AMQ Streams is available for download from the Red Hat website.
```
Comment (scholzj) [inline books/proc-downloading-kafka-bridge.adoc]:
```suggestion
A zipped distribution of AMQ Streams is available for download from the Red Hat website.
```

File: books/proc-installing-kafka-bridge.adoc [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+.Prerequisites
+
+* xref:proc-installing-amq-streams-{context}[{ProductName} is installed on the host]
+* xref:proc-downloading-kafka-bridge-{context}[The Kafka Bridge installation archive is downloaded]
+* xref:proc-configuring-kafka-bridge-{context}[The Kafka Bridge configuration properties have been reviewed]
+
+.Procedure
+
+. Run the installation script provided with the {ProductName} Kafka Bridge installation archive:
```
Comment (scholzj) [inline books/proc-installing-kafka-bridge.adoc]: There is AFAIK no installation script - they just have to unzip the archive.

File: books/proc-installing-kafka-bridge.adoc [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+[source,shell]
+strimzi-kafka-bridge/bin/kafka_bridge_run.sh
++
+Use the configuration properties as a parameter.
++
+For example:
++
+[source,shell]
+----
+./kafka_bridge_run.sh --config-file=_path_/configfile.properties
```
Comment (scholzj) [inline books/proc-installing-kafka-bridge.adoc]:
```suggestion
./bin/kafka_bridge_run.sh --config-file=_path_/configfile.properties
```

File: books/ref-api-resources-kafka-bridge.adoc [hunk 14]
```diff
@@ -0,0 +1,8 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-kafka-bridge.adoc
+
+[id='ref-api-resources-kafka-bridge-{context}']
+= {ProductName} Kafka Bridge API resources
+
+Content goes here.
```
Comment (scholzj) [inline books/ref-api-resources-kafka-bridge.adoc]: Will you link here? Or will you just include the resources?

File: books/ref-overview-supported-configurations.adoc [hunk 15]
```diff
@@ -20,5 +20,5 @@ In order to be running in a supported configuration, AMQ Streams must be running
 [width="80%",cols="3,^2,^2",options="header"]
 |=========================================================
 |Operating System |Architecture |Version
-|Red Hat Enterprise Linux |x86_64 |7.x
+|{ProductPlatformName} |x86_64 |7.x
```
Comment (scholzj) [inline books/ref-overview-supported-configurations.adoc]: This might be for another PR. But AFAIU we will need to add RHEL8 here as well.

File: books/con-overview-of-kafka-bridge.adoc (hunk 2 above)
Comment (PaulRMellor) [reply books/con-overview-of-kafka-bridge.adoc]: The resources relate only to the Kafka Bridge, not the consumers and producers connected directly to Kafka.

File: books/con-requests-kafka-bridge.adoc (hunk 4 above)
Comment (PaulRMellor) [reply books/con-requests-kafka-bridge.adoc]: @scholzj  We can add a bit more here, if there's something specific to the bridge. Can you explain the prefix with `kafka` a bit more?

File: books/con-sending-requests-to-kafka-bridge.adoc (hunk 5 above)
Comment (PaulRMellor) [reply books/con-sending-requests-to-kafka-bridge.adoc]: It's out. It was going to track the OCP, but it's out of that too.

Comment (laidan6000) [reply books/con-sending-requests-to-kafka-bridge.adoc]: I've removed the whole concept from #1668. It can be created in a future PR.

File: books/ref-api-resources-kafka-bridge.adoc (hunk 14 above)
Comment (laidan6000) [reply books/ref-api-resources-kafka-bridge.adoc]: I've updated the content for this module in #1668.

"For the full list of REST API endpoints and descriptions, including example requests and responses, see [Kafka Bridge API reference](https://strimzi.io/docs/bridge/0.12.0/full.html)."

File: books/con-requests-kafka-bridge.adoc [hunk 16]
```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-kafka-bridge.adoc
+
+[id='con-requests-kafka-bridge-{context}']
+= Requests to the {ProductName} Kafka Bridge
```
Comment (laidan6000) [inline books/con-requests-kafka-bridge.adoc]: I made substantial edits to this module in #1668 on 07/06/19. Please update this section to match.

File: books/ref-api-resources-kafka-bridge.adoc (hunk 14 above)
Comment (PaulRMellor) [reply books/ref-api-resources-kafka-bridge.adoc]: Link is in now.

File: books/ref-overview-supported-configurations.adoc (hunk 15 above)
Comment (PaulRMellor) [reply books/ref-overview-supported-configurations.adoc]: I can add another line. What do we want?

File: books/con-requests-kafka-bridge.adoc (hunk 4 above)
Comment (laidan6000) [reply books/con-requests-kafka-bridge.adoc]: Does this comment also apply to the Bridge on OCP docs, @scholzj ?

File: books/con-sending-requests-to-kafka-bridge.adoc (hunk 5 above)
Comment (laidan6000) [reply books/con-sending-requests-to-kafka-bridge.adoc]: Please remove the module. It has been removed in #1668

File: books/proc-configuring-kafka-bridge.adoc (hunk 6 above)
Comment (laidan6000) [inline books/proc-configuring-kafka-bridge.adoc]: Duplicated word: the the

File: books/proc-configuring-kafka-bridge.adoc [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+
+.Procedure
+
+. Edit the the `configfile.properties` file provided with the {ProductName} Kafka Bridge installation archive.
++
+Use the properties file to specify Kafka and HTTP-related properties.
+
+.. Configure standard Kafka-related properties, including properties specific to the Kafka consumers and producers.
++
+---
```
Comment (laidan6000) [inline books/proc-configuring-kafka-bridge.adoc]: What is this line for?

File: books/proc-configuring-kafka-bridge.adoc [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+Use:
+* `kafka.bootstrap.servers` to define the host/port connections to the Kafka cluster
+* `kafka.producer.acks` to provide acknowledgments to the HTTP client
+* `kafka.consumer.auto.offset.reset`’ to determine how to manage reset of the offset in Kafka
```
Comment (laidan6000) [inline books/proc-configuring-kafka-bridge.adoc]: Remove the rogue apostrophe from the third bullet point.

File: books/proc-configuring-kafka-bridge.adoc [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+---
```
Comment (laidan6000) [inline books/proc-configuring-kafka-bridge.adoc]: Another `---` here?

File: books/proc-installing-kafka-bridge.adoc [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+[id='proc-installing-kafka-bridge{context}']
+= Installing the {ProductName} Kafka Bridge on {ProductPlatformName}
+
+Follow this procedure to install the {ProductName} Kafka Bridge on {ProductPlatformName}.
+
+.Prerequisites
+
+* xref:proc-installing-amq-streams-{context}[{ProductName} is installed on the host]
+* xref:proc-downloading-kafka-bridge-{context}[The Kafka Bridge installation archive is downloaded]
+* xref:proc-configuring-kafka-bridge-{context}[The Kafka Bridge configuration properties have been reviewed]
```
Comment (laidan6000) [inline books/proc-installing-kafka-bridge.adoc]: Have been set rather than reviewed?

File: books/proc-installing-kafka-bridge.adoc (hunk 12 above)
Comment (PaulRMellor) [reply books/proc-installing-kafka-bridge.adoc]: Just spoken with @stanlyDoge to confirm and updated accordingly.

File: books/con-requests-kafka-bridge.adoc (hunk 16 above)
Comment (PaulRMellor) [reply books/con-requests-kafka-bridge.adoc]: We're in sync

File: books/proc-configuring-kafka-bridge.adoc (hunk 17 above)
Comment (PaulRMellor) [reply books/proc-configuring-kafka-bridge.adoc]: It wraps up formatting, so you don't need to keep adding '+' signs.

File: books/proc-configuring-kafka-bridge.adoc (hunk 19 above)
Comment (PaulRMellor) [reply books/proc-configuring-kafka-bridge.adoc]: end of wrap

File: books/ref-overview-supported-configurations.adoc (hunk 15 above)
Comment (scholzj) [reply books/ref-overview-supported-configurations.adoc]: I think it should be `|{ProductPlatformName} |x86_64 |8.x`

Comment (scholzj) [reply books/ref-overview-supported-configurations.adoc]: Or just edit the original one to say `7.x, 8.x`

File: books/con-requests-kafka-bridge.adoc (hunk 4 above)
Comment (scholzj) [reply books/con-requests-kafka-bridge.adoc]: @laidan6000 On OCP the security is configured b the operator. How to configure it is described in the part of docs prepared by Stanislav.

@PaulRMellor Basically, in the configuration file users can use 3 different rpefixes to configure things:
* `kafka.` should be used for general configuration which would apply to both producer and consumer. It will be always applied to both. The common configuration is for example the sever where to connect and the security.
* `kafka.consumer.` should be used for consumer specific configuration. These will be passed only to the consumer.
* `kafka.producer.` should be used for producer specific configuration. These will be passed only to the producer.

And when the same option is specified with `kafka.` and `kafka.consumer.` or `kafka.producer.`, the one with the longer prefix will always be used primarily.

Comment (PaulRMellor) [reply books/con-requests-kafka-bridge.adoc]: Thanks for clarifying Jakub. I have dropped out that link and added a line to `con-requests-kafka-bridge.adoc` with a link to info on the prefixes to 'proc-configuring-kafka-bridge.adoc'

File: books/ref-overview-supported-configurations.adoc (hunk 15 above)
Comment (PaulRMellor) [reply books/ref-overview-supported-configurations.adoc]: Gone for the single line. What about:

.List of supported Java Virtual Machines
[width="80%",cols="3,^2",options="header"]
|=========================================================
|Java Virtual Machine |Version
|OpenJDK |1.8
|OracleJDK |1.8
|IBM JDK |1.8
|=========================================================

File: books/con-requests-kafka-bridge.adoc [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+  "format": "binary", <1>
+...
+}
+----
+
+<1> A binary embedded data format.
+
+The embedded data format specified when creating a consumer must match the data format of the Kafka messages it will consume.
+
+If you choose to specify a binary embedded data format, subsequent producer requests must provide the binary data in the request body as Base64-encoded strings. For example, when sending messages by making `POST` requests to the `/topics/_topicname_` endpoint, the `value` must be in Base64:
```
Comment (ppatierno) [inline books/con-requests-kafka-bridge.adoc]: "must be encoded in Base64"

File: books/ref-overview-supported-configurations.adoc (hunk 15 above)
Comment (PaulRMellor) [reply books/ref-overview-supported-configurations.adoc]: .List of supported Java Virtual Machines
[width="80%",cols="3,^2",options="header"]
|=========================================================
|Java Virtual Machine |Version
|OpenJDK |1.8, 11 footnote:[For RHEL 8.x]
|OracleJDK |1.8
|IBM JDK |1.8
|=========================================================
